src/main/java/seedu/address/logic/commands/MarkTaskCommand.java

```diff
@@ -0,0 +1,53 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.task.Task;
+
+/**
+ * Marks the task at the specified index to be completed.
+ */
+public class MarkTaskCommand extends Command {
+    public static final String COMMAND_WORD = "marktask";
+
+    public static final String MESSAGE_MARK_TASK_SUCCESS = "Task at %1$s set as completed!";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Marks the task at specified index in the displayed task list.\n"
+            + "Parameters: INDEX (must be a positive integer)\n"
+            + "Example: " + COMMAND_WORD + " 1";
+
+    private final Index targetIndex;
+
+    public MarkTaskCommand(Index index) {
+        this.targetIndex = index;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Task> lastShownList = model.getFilteredTaskList();
+
+        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_TASK_DISPLAYED_INDEX);
+        }
+
+        model.markTask(targetIndex);
```

Comment:
I was wondering if we would allow users to mark an already marked task/ unmark an already unmarked task, since the code does not check for that.